src/integrations/ga/GaToSplit.js

```diff
@@ -1,7 +1,257 @@
+  if (properties === false)
+    return false;
+
+  if (data.timestamp && !isNumber(data.timestamp))
+    return false;
+
+  if (data.key && validateKey(data.key, 'splitio-ga-to-split:mapper') === false)
+    return false;
+
+  if (data.trafficTypeName && validateTrafficType(data.trafficTypeName, 'splitio-ga-to-split:mapper') === false)
+    return false;
+
+  return true;
+}
+
+/**
+ * GaToSplit integration.
+ * This function provides the SplitTracker plugin to ga command queue.
+ *
+ * @param {object} sdkOptions options passed at the SDK integrations settings
+ * @param {object} storage SDK storage passed to track events
+ * @param {object} coreSettings core settings used to define an identity if no one provided as SDK or plugin options
+ */
+function GaToSplit(sdkOptions, storage, coreSettings) {
+
+  const defaultOptions = {
+    filter: defaultFilter,
+    mapper: defaultMapper,
+    // Default eventHandler stores event
+    eventHandler: function (event) {
+      storage.events.track(event);
+    },
+    // We set default identities if key and TT are present in settings.core
+    identities: (coreSettings.key && coreSettings.trafficType) ?
+      [{ key: coreSettings.key, trafficType: coreSettings.trafficType }] :
+      undefined
+  };
+
+  class SplitTracker {
+
+    // Constructor for the SplitTracker plugin.
+    constructor(tracker, pluginOptions) {
+
+      // precedence of options: SDK options (config.integrations) overwrite pluginOptions (`ga('require', 'splitTracker', pluginOptions)`)
+      const opts = Object.assign({}, defaultOptions, sdkOptions, pluginOptions);
+
+      this.tracker = tracker;
+
+      // Validate identities
+      const validIdentities = validateIdentities(opts.identities);
+
+      if (!validIdentities || validIdentities.length === 0) {
+        log.warn('No valid identities were provided. Please check that you are passing a valid list of identities or providing a traffic type at the SDK configuration.');
+        return;
+      }
+
+      const invalids = validIdentities.length - opts.identities.length;
+      if (invalids) {
+        log.warn(`${invalids} identities were discarded because they are invalid or duplicated. Identities must be an array of objects with key and trafficType.`);
+      }
+      opts.identities = validIdentities;
+
+      // Validate prefix
+      // @TODO Improve the prefix validation using the same REGEX than eventTypeId
+      if (opts.prefix && !isString(opts.prefix)) {
+        log.warn('The provided `prefix` was ignored since it is invalid. Please check that you are passing a string object as `prefix`.');
+        opts.prefix = undefined;
+      }
+
+      // Overwrite sendHitTask to perform plugin tasks:
+      // 1) filter hits
+      // 2) map hits to Split events
+      // 3) handle events, i.e., validate and send them to Split BE
+      const originalSendHitTask = tracker.get('sendHitTask');
+      tracker.set('sendHitTask', function (model) {
```

Comment:
Maybe we could move this function body, but I'm not requesting this change.

Having options and tracker on the instance should make it possible. Might be another option to use another function to also overwrite the task so you keep the reference in that context, while accessing tracker and options from this. (will need to bind the context for the callback though).